addons/account/models/account_move.py

```diff
@@ -3130,17 +3156,55 @@ def _compute_has_reconciled_entries(self):
-        ])
-        for ids in self._cr.split_for_in_conditions(records.ids, size=1000):
-            self.browse(ids)._post()
-            if not self.env.registry.in_test_mode():
-                self._cr.commit()
+            ('auto_post', '!=', 'no'),
+        ], limit=100)
+        records._post()
+        records.filtered(lambda r: r.auto_post != 'at_date')._copy_recurring_entries()
+
+        if len(records) == 100:  # assumes there are more whenever search hits limit
+            self.env.ref('account.ir_cron_auto_post_draft_entry')._trigger()
+
+    def _copy_recurring_entries(self):
+        ''' Creates a copy of a recurring (periodic) entry and adjusts its dates for the next period.
+        Meant to be called right after posting a periodic entry.
+        Copies extra fields as defined by get_fields_to_copy_recurring_entries().
+        '''
+        def apply_delta(date, date_origin, months):
+            # Advances date by some months, maintaining original day of the month if possible
+            delta = relativedelta(months=months)  # does not overflow when lands on shorter month
+            next_date = date + delta
+            next_month_days = monthrange(next_date.year, next_date.month)[1]
+            origin_day = date_origin.day
+            return next_date + relativedelta(days=min(next_month_days, origin_day) - next_date.day)
+
+        deltas = {'monthly': 1, 'quarterly': 3, 'yearly': 12}
+        for record in self:
+            record.auto_post_origin_id = record.auto_post_origin_id or record  # original entry references itself
+            next_date = apply_delta(record.date, record.auto_post_origin_id.date, deltas[record.auto_post])
+
+            if not record.auto_post_until or next_date <= record.auto_post_until:  # recurrence continues
+                copy = record.copy(default=record.get_fields_to_copy_recurring_entries())
+                copy.date = next_date
+                copy.invoice_date = apply_delta(record.invoice_date, record.auto_post_origin_id.invoice_date, deltas[record.auto_post])
+                if not copy.invoice_payment_term_id:  # no payment terms: maintain timedelta between due date and accounting date
+                    copy.invoice_date_due = copy.date + (record.invoice_date_due - record.date)
+
+    def get_fields_to_copy_recurring_entries(self):
+        ''' Determines which extra fields to copy when copying a recurring entry.
+        To be extended by modules that add fields with copy=False (implicit or explicit)
+        whenever the opposite behavior is expected for recurring invoices.
+        '''
+        return {
+            'auto_post': self.auto_post,  # copy=False to avoid mistakes but should be the same in recurring copies
+            'auto_post_until': self.auto_post_until,  # same as above
+            'auto_post_origin_id': self.auto_post_origin_id.id,  # same as above
+            'invoice_user_id': self.invoice_user_id.id,  # otherwise user would be OdooBot
```

Comment:
that's where you need to set the right date, invoice_date, invoice_date_due...

Reply:
> Les gouts et les couleurs ne se discutent pas

But it feels weird to me to compute the `date` in `_copy_recurring_entries` (it has to be known at that time) and the `invoice_date` in `_get_fields_to_copy_recurring_entries`